packages/expo-updates/android/src/androidTest/java/expo/modules/updates/selectionpolicy/ReaperSelectionPolicyDevelopmentClientTest.java

```diff
@@ -0,0 +1,108 @@
+    update3.lastAccessed = new Date(1619569813253L);
+    update4.lastAccessed = new Date(1619569813254L);
+    update5.lastAccessed = new Date(1619569813255L);
+
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(
+            // the order of the list shouldn't matter
+            Arrays.asList(update2, update5, update4, update1, update3),
+            update5,
+            null);
+
+    Assert.assertEquals(2, updatesToDelete.size());
+    Assert.assertTrue(updatesToDelete.contains(update1));
+    Assert.assertTrue(updatesToDelete.contains(update2));
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_SameLastAccessedDate() {
+    // if multiple updates have the same lastAccessed date, should use commitTime to determine
+    // which updates to delete
+    update1.lastAccessed = new Date(1619569813250L);
+    update2.lastAccessed = new Date(1619569813250L);
+    update3.lastAccessed = new Date(1619569813250L);
+    update4.lastAccessed = new Date(1619569813250L);
+
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(
+            Arrays.asList(update3, update4, update1, update2),
+            update4,
+            null);
+
+    Assert.assertEquals(1, updatesToDelete.size());
+    Assert.assertTrue(updatesToDelete.contains(update1));
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_LaunchedUpdateIsOldest() {
+    // if the least recently accessed update happens to be launchedUpdate, delete instead the next
+    // least recently accessed update
+    update1.lastAccessed = new Date(1619569813251L);
+    update2.lastAccessed = new Date(1619569813252L);
+    update3.lastAccessed = new Date(1619569813253L);
+    update4.lastAccessed = new Date(1619569813254L);
+
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(
+            Arrays.asList(update1, update2, update3, update4),
+            update1,
+            null);
+
+    Assert.assertEquals(1, updatesToDelete.size());
+    Assert.assertTrue(updatesToDelete.contains(update2));
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_NoLaunchedUpdate() {
+    // if launchedUpdate is null, something has gone wrong, so don't delete anything
+    List<UpdateEntity> updatesToDelete = selectionPolicy.selectUpdatesToDelete(
+            Arrays.asList(update1, update2, update3, update4),
+            null,
+            null);
+    Assert.assertEquals(0, updatesToDelete.size());
+  }
+
+  @Test
+  public void testSelectUpdatesToDelete_BelowMaxNumber() {
+    // no need to delete any updates if we have <= the max number of updates
+    Assert.assertEquals(0, selectionPolicy.selectUpdatesToDelete(
+            Arrays.asList(update1, update2),
+            update2,
+            null).size());
+    Assert.assertEquals(0, selectionPolicy.selectUpdatesToDelete(
+            Arrays.asList(update1, update2, update3),
+            update3,
+            null).size());
+  }
```

Comment:
Right now it will require a bunch of mocking; will see if I can get something working...

Reply:
Ok, added tests for this on both platforms. It might also be a good task for either you or I sometime this cycle to clean up the Launcher/Loader classes a little bit and make them more testable; right now it's rather messy (some of which is inherent since unit testing these classes requires mocking) but would be great to have more tests since these are some of the more critical/brittle pieces of the codebase.